simulator/mips/mips_instr.cpp

```diff
@@ -216,12 +226,27 @@ FuncInstr::FuncInstr( uint32 bytes, Addr PC,
 void FuncInstr::initFormat()
 {
     bool is_R = ( instr.asR.opcode == 0x0);
+    bool is_RI = ( instr.asR.opcode == 0x1);
     uint8 ident = is_R ? instr.asR.funct : instr.asR.opcode;
 
-    for ( const auto& entry : isaTable)
+    if ( is_R)
+    {    
+        const auto& entry = isaMapR.at( ident);
+        if ( entry != isaMapR.end())
+        {
+            format    = entry.format;
+            operation = entry.operation;
+            mem_size  = entry.mem_size;
+            name      = entry.name.c_str();
+            function  = entry.function;
+            return;
+        }	 
+    }
+    
+    else if( is_RI)
     {
-        bool is_i_R = ( entry.format == FORMAT_R);
-        if ( entry.opcode == ident && ( is_i_R == is_R))
+        const auto& entry = isaMapRI.at( ident);
+        if ( entry != isaMapRI.end())
```

Comment:
`std::unordered_map::at` returns `const FuncInstr::ISAEntry&`, which cannot be compared to `isaMapRI.end()`. Have you tried to build your program before sending it to review?

Reply:
I did not figure out how to test this module

Reply:
Ok, I will correct it

Reply:
What do you mean by "I did not figure out"?? First of all, there is a README file which explains how to build a simulator. Secondly, there is my e-mail, where you can get answers for your questions.

Reply:
When I builded simulator, I got error in file infra/cache/cache_tag_array.cpp, so I couldn't check my changes

Reply:
So you faced the problem and decided that it is all right, without any attempts to solve it (write to me or check that compiler version should be GCC 7 at least in README), right? That is not the best way to solve problems :-)

Reply:
Compiler version is GCC 7, I was wrong, sorry

Reply:
Please send your command line and an error output to me in e-mail.